File: 3731-find-missing-elements/3731-find-missing-elements.java
class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        List<Integer> res = new ArrayList();
        int mn = Arrays.stream(nums).min().getAsInt();
        int mx = Arrays.stream(nums).max().getAsInt();
        Set<Integer> st = new HashSet();
        for(int el : nums)  st.add(el);
        for(int i = mn ; i <= mx ; i++){
            if(!st.contains(i)){
                res.add(i);
            }
        }
        return res;
    }
}

File: 3731-find-missing-elements/3731-find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        n = max(nums)
        p = min(nums)
        res = []
        for i in range(p,n+1):
            if i not in nums:
                res.append(i)
        return res